CHANGELOG.md

```diff
@@ -2,6 +2,35 @@
 
 This document includes a curated changelog for each release. We also publish a changelog as the description of a [GitHub release](https://github.com/nginxinc/nginx-kubernetes-gateway/releases), which, by contrast, is auto-generated and includes links to all PRs that went into the release.
 
+## Release 0.5.0
+
+*July 17, 2023*
+
+This release brings:
+- Support for all v1beta1 Core features of the Gateway API resources. See the [Gateway Compatibility doc](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.5.0/docs/gateway-api-compatibility.md)
+- Miscellaneous enhancements and bug fixes.
```

Comment:
I wonder if it adds any value here - cause those enhancements and bug fixes are mostly about conformance (the list element above)

perhaps we don't even need a list?

---

This release completes support for all v1beta1 Core features of the Gateway API resources. See the [Gateway Compatibility doc](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.5.0/docs/gateway-api-compatibility.md).

---

Maybe completes better than brings, because we added a few missing features rather than all v1beta1 core features